src/display/api.js

```diff
@@ -2646,32 +2646,27 @@ class WorkerTransport {
       };
     });
 
-    messageHandler.on("ReaderHeadersReady", data => {
-      const headersCapability = Promise.withResolvers();
-      const fullReader = this._fullReader;
-      fullReader.headersReady.then(() => {
-        // If stream or range are disabled, it's our only way to report
-        // loading progress.
-        if (!fullReader.isStreamingSupported || !fullReader.isRangeSupported) {
-          if (this._lastProgress) {
-            loadingTask.onProgress?.(this._lastProgress);
-          }
-          fullReader.onProgress = evt => {
-            loadingTask.onProgress?.({
-              loaded: evt.loaded,
-              total: evt.total,
-            });
-          };
-        }
+    messageHandler.on("ReaderHeadersReady", async data => {
+      await this._fullReader.headersReady;
```

Comment:
Just an idea: could we just have a getter/function in order to have something like `const { isStreamingSupported, ...} = await this._fullReader.FooBar()` ?

Reply:
That'd be possible, however after considering it I'd prefer not to given the amount of code duplication that it'd lead to. Please note that we have no less than *four* different `IPDFStreamReader` implementations:
 - https://github.com/mozilla/pdf.js/blob/master/src/display/fetch_stream.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/network.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/node_stream.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/transport_stream.js